bbb-common-web/src/main/scala/org/bigbluebutton/api2/util/RecordingMetaXmlHelper.scala

```diff
@@ -188,19 +190,35 @@ class RecMetaXmlHelper extends RecordingServiceGW with LogHelper {
     }
   }
 
-  def getRecordingTextTracks(recordId: String, captionsDir: String): String = {
+  def getRecordingTextTracks(recordId: String, captionsDir: String, captionBasUrl: String): String = {
     val gson = new Gson()
     var returnResponse: String = ""
     val captionsFilePath = captionsDir + File.separatorChar + recordId + File.separatorChar + CAPTIONS_FILE
 
     readCaptionJsonFile(captionsFilePath, StandardCharsets.UTF_8) match {
       case Some(captions) =>
-        val ctracks = gson.fromJson(captions, classOf[util.ArrayList[Track]])
-        val result1 = GetRecTextTracksResult(SUCCESS, ctracks)
-        val response1 = GetRecTextTracksResp(result1)
-        val respText1 = gson.toJson(response1)
+        val ctracks = gson.fromJson(captions, classOf[java.util.List[LinkedTreeMap[String, String]]])
+
+        val list = new util.ArrayList[Track]()
+        val it = ctracks.iterator()
+
+        while (it.hasNext()) {
+          val mapTrack = it.next()
+          list.add(new Track(
+            href = captionBasUrl + mapTrack.get("lang") + ".json",

```

Comment:
The caption filenames will have the extension `.vtt` not `.json`

We're going to change this later to provide authenticated/signed URLs to fetch the caption files.